components/x-topic-search/src/TopicSearch.jsx

```diff
@@ -33,6 +36,22 @@ const topicSearchActions = withActions(({ minSearchLength = 2, maxSuggestions =
 		}
 	},
 
+	topicFollowed (subjectId, subjectName) {
+		const followedTopicIndex = getFollowedTopicIndex(followedTopics, subjectId);
+
+		if (followedTopicIndex === -1) {
+			followedTopics.push({ name: subjectName, uuid: subjectId })
+		}
+	},
+
+	topicUnfollowed (subjectId) {
+		const unfollowedTopicIndex = getFollowedTopicIndex(followedTopics, subjectId);
+
+		if (unfollowedTopicIndex > 0) {
+			delete followedTopics[ unfollowedTopicIndex ];
```

Comment:
`delete` on an array will leave an empty element - it won't reindex the array or affect the array `.length`

`splice` does the job properly:

```suggestion
			followedTopics.splice(unfollowedTopicIndex, 1);
```

Reply:
I see!! Thank you!